### ISSUE TITLE
Error: Type '(typeof Counter__SvelteComponent_)[]' is not assignable to type 'SvelteComponentDev[]'

### ISSUE THREAD

**Issue by acarl005** [sveltejs/language-tools#1616]
### Describe the bug

An array of svelte components is not able to be assigned to a type of `Array<SvelteComponent>`. I imported like so...

```svelte
<script lang="ts">
  import type { SvelteComponent } from "svelte"

  export let components: Array<SvelteComponent>
  // ...
</script>
```

However, when I try to actually use this component and pass that prop, I get this error from typescript:

```
/home/projects/sveltejs-kit-template-default-8unn5h/src/routes/+page.svelte:32:3
Error: Type '(typeof Counter__SvelteComponent_)[]' is not assignable to type 'SvelteComponentDev[]'.
  Type 'typeof Counter__SvelteComponent_' is missing the following properties from type 'SvelteComponentDev': $set, $on, $destroy, $$prop_def, and 5 more. (ts)
                names={tabNames}
                components={tabComponents}
        />


====================================
svelte-check found 1 error, 0 warnings, and 0 hints
```

You can go to my [reproduction](https://stackblitz.com/edit/sveltejs-kit-template-default-8unn5h?file=src%2Froutes%2F%2Bpage.svelte,src%2Froutes%2FTabs.svelte&terminal=dev) and run `npm run check`.

### Reproduction

https://stackblitz.com/edit/sveltejs-kit-template-default-8unn5h?file=src%2Froutes%2F%2Bpage.svelte,src%2Froutes%2FTabs.svelte&terminal=dev

```sh
npm run check
```

### Expected behaviour

The types should be valid, but instead I get a type error about an array of svelte components.

### System Info

```sh
npx envinfo --system --binaries --browsers --npmPackages "{svelte,@sveltejs/*,vite,typescript,svelte-check}"
```

```
  System:
    OS: Linux 5.0 undefined
    CPU: (4) x64 Intel(R) Core(TM) i9-9880H CPU @ 2.30GHz
    Memory: 0 Bytes / 0 Bytes
    Shell: 1.0 - /bin/jsh
  Binaries:
    Node: 16.14.2 - /usr/local/bin/node
    Yarn: 1.22.10 - /usr/local/bin/yarn
    npm: 7.17.0 - /usr/local/bin/npm
  npmPackages:
    @sveltejs/adapter-auto: next => 1.0.0-next.71 
    @sveltejs/kit: next => 1.0.0-next.463 
    svelte: ^3.46.0 => 3.49.0 
    svelte-check: ^2.7.1 => 2.9.0 
    typescript: ^4.7.4 => 4.8.2 
    vite: ^3.1.0-beta.1 => 3.1.0-beta.1 
```

### Which package is the issue about?

svelte-check

**Comment by jasonlyu123** [comment]
The error is correct. What you want is the class constructor of the component but you typed as the instance of the class. 

There's a new utility type `ComponentType` released in svelte 3.49 you can use to extract the constructor. For example

```ts
import type { ComponentType, SvelteComponentTyped } from 'svelte';
export let components: ComponentType<SvelteComponentTyped<{ needsThisProp: string }>>[]
```

**Comment by acarl005** [comment]
@jasonlyu123 Thanks for the help :)
